app/integration/runtime/src/test/java/io/syndesis/integration/runtime/handlers/TemplateStepHandlerTest.java

```diff
@@ -0,0 +1,363 @@
+                                                        .replace(symbols[0].id, "10:00")
+                                                        .replace(symbols[1].id, "Bob")
+                                                        .replace(symbols[2].id, "Greetings, How are you?");
+        expectedMessages.add(toJson(expectedMessage));
+
+        expectedMessage = template.replaceAll("\\{\\{|\\}\\}", EMPTY_STRING)
+                                                         .replace(symbols[0].id, "10:15")
+                                                         .replace(symbols[1].id, "Susan")
+                                                         .replace(symbols[2].id, "Hello, Pleased to meet you!");
+        expectedMessages.add(toJson(expectedMessage));
+
+        try {
+            final RouteBuilder routes = generateRoute(template, Arrays.asList(symbols));
+
+            // Set up the camel context
+            context.addRoutes(routes);
+            dumpRoutes(context);
+
+            context.start();
+
+            // Dump routes as XML for troubleshooting
+            dumpRoutes(context);
+
+            MockEndpoint result = context.getEndpoint("mock:result", MockEndpoint.class);
+            result.setExpectedMessageCount(2);
+            result.expectedBodiesReceived(expectedMessages);
+
+            sendData(context, testMessages);
+
+            result.assertIsSatisfied();
+
+        } finally {
+            context.stop();
+        }
+     }
+
+    @Test
+    public void testTemplateStepIterate() throws Exception {
+        CamelContext context = new SpringCamelContext(applicationContext);
+
+        List<String> testMessages = new ArrayList<>();
+        testMessages.add(message(snippet("name", "Bob"),
+                                                             snippet("course",
+                                                                 snippet("name", "Mathematics"),
+                                                                 snippet("name", "English"),
+                                                                 snippet("name", "Chemistry")
+                                                             ),
+                                                             snippet("text", "Will be going to the University of Southampton.")));
+
+        testMessages.add(message(snippet("name", "Susan"),
+                                                             snippet("course",
+                                                                   snippet("name", "Physics"),
+                                                                   snippet("name", "German"),
+                                                                   snippet("name", "Art")
+                                                             ),
+                                                             snippet("text", "Will be joining Evans, Richards and Dean.")));
```

Comment:
can we reduce a little this indentation ? it makes it hard to review as the line goes beyond github limits. Of course not a blocking issue :)

Reply:
+1